src/twisted/web/iweb.py

```diff
@@ -82,11 +82,30 @@ def getRequestHostname():
         """
         Get the hostname that the user passed in to the request.
 
-        This will either use the Host: header (if it is available) or the
-        host we are listening on if the header is unavailable.
+        This will either use the Host: header (if it is available) or
+        the host we are listening on if the header is unavailable.
+
+        This method is B{deprecated}.  See L{getRequestHost} instead.
 
         @returns: the requested hostname
-        @rtype: C{str}
+
+        @rtype: L{str}
+        """
+
+
+    def getRequestHost():
```

Comment:
Isn't adding a method to an interface a backwards-incompatible change?

Reply:
Agh, yes, it is - I suppose this will require a new interface.

Reply:
Since getRequestHost is used for now only in the existing getRequestHostname, I think that is safe to leave it out of the interface and just handle it like an implementation detail.

Do we need it as a public method?

Maybe we can create a list of methods for IRequestV2 :) which can include the required methods to allow traversal before all content is received (after all headers are received) and deferred traversal.